common/src/main/java/com/wynntils/features/inventory/UnidentifiedItemIconFeature.java

```diff
@@ -11,39 +11,62 @@
+import java.util.function.Function;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
 import net.minecraft.world.item.ItemStack;
 import net.neoforged.bus.api.SubscribeEvent;
 
 @ConfigCategory(Category.INVENTORY)
 public class UnidentifiedItemIconFeature extends Feature {
-    private static final Map<GearType, Pair<Integer, Integer>> TEXTURE_COORDS = Map.ofEntries(
-            Map.entry(GearType.SPEAR, Pair.of(16 * 1, 16 * 1)),
-            Map.entry(GearType.WAND, Pair.of(16 * 0, 16 * 1)),
-            Map.entry(GearType.DAGGER, Pair.of(16 * 2, 16 * 1)),
-            Map.entry(GearType.BOW, Pair.of(16 * 3, 16 * 1)),
-            Map.entry(GearType.RELIK, Pair.of(16 * 0, 16 * 2)),
-            Map.entry(GearType.RING, Pair.of(16 * 1, 16 * 2)),
-            Map.entry(GearType.BRACELET, Pair.of(16 * 2, 16 * 2)),
-            Map.entry(GearType.NECKLACE, Pair.of(16 * 3, 16 * 2)),
-            Map.entry(GearType.HELMET, Pair.of(16 * 0, 16 * 0)),
-            Map.entry(GearType.CHESTPLATE, Pair.of(16 * 1, 16 * 0)),
-            Map.entry(GearType.LEGGINGS, Pair.of(16 * 2, 16 * 0)),
-            Map.entry(GearType.BOOTS, Pair.of(16 * 3, 16 * 0)),
-            Map.entry(GearType.MASTERY_TOME, Pair.of(16 * 0, 16 * 3)),
-            Map.entry(GearType.CHARM, Pair.of(16 * 1, 16 * 3)));
+    private static final StyledText QUESTION_MARK_TEXT =
+            StyledText.fromComponent(Component.literal("?").withStyle(ChatFormatting.BOLD));
+    private static final Map<GearType, Pair<Integer, Integer>> TEXTURE_COORDS = new EnumMap<>(GearType.class);
+
+    static {
+        TEXTURE_COORDS.put(GearType.SPEAR, Pair.of(16 * 1, 16 * 1));
+        TEXTURE_COORDS.put(GearType.WAND, Pair.of(16 * 0, 16 * 1));
+        TEXTURE_COORDS.put(GearType.DAGGER, Pair.of(16 * 2, 16 * 1));
+        TEXTURE_COORDS.put(GearType.BOW, Pair.of(16 * 3, 16 * 1));
+        TEXTURE_COORDS.put(GearType.RELIK, Pair.of(16 * 0, 16 * 2));
+        TEXTURE_COORDS.put(GearType.RING, Pair.of(16 * 1, 16 * 2));
+        TEXTURE_COORDS.put(GearType.BRACELET, Pair.of(16 * 2, 16 * 2));
+        TEXTURE_COORDS.put(GearType.NECKLACE, Pair.of(16 * 3, 16 * 2));
+        TEXTURE_COORDS.put(GearType.HELMET, Pair.of(16 * 0, 16 * 0));
+        TEXTURE_COORDS.put(GearType.CHESTPLATE, Pair.of(16 * 1, 16 * 0));
+        TEXTURE_COORDS.put(GearType.LEGGINGS, Pair.of(16 * 2, 16 * 0));
+        TEXTURE_COORDS.put(GearType.BOOTS, Pair.of(16 * 3, 16 * 0));
+        TEXTURE_COORDS.put(GearType.MASTERY_TOME, Pair.of(16 * 0, 16 * 3));
+        TEXTURE_COORDS.put(GearType.CHARM, Pair.of(16 * 1, 16 * 3));
+    }
 
     @Persisted
     public final Config<UnidentifiedItemTextures> texture = new Config<>(UnidentifiedItemTextures.WYNN);
 
+    @Persisted
+    public final Config<Boolean> showOnUnboxed = new Config<>(true);
+
+    @Persisted
+    public final Config<UnidentifiedItemTextures> unboxedTexture = new Config<>(UnidentifiedItemTextures.QUESTION_MARK);
```

Comment:
I don't like how you mix the old UnidentifiedItemTextures with this new function. They serve quite different purposes. For boxed items, the texture draws a "mini object" on top of the box, and the reason we have UnidentifiedItemTextures is to allow users to select between two different themes, one which mimicks Wynn style better, and one with clearer legibility in the small size.

The marker for "unboxed but unidentified" items is very different -- in this case it would make no sense to overlay an icon of what type it is, since the real icon is clearly visible. And in contrast, the question mark would make no sense to put on all boxed items.

I think the question mark is fine as it is -- you don't need to treat it as a texture, but only present the option to draw a question mark or not. If someone at a later stage gets mad at the question mark and wants to provide a different texture, then that is a future refactoring to allow different textures for "unboxed but un-id:d" items.

Reply:
Doing this will also allow you to revert several changes to this file, and just add a new event listener that draws a question mark on top of unboxed-but-unidentified objects.